subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -3337,6 +3342,9 @@ static int do_send_reply_cb(const struct coap_packet *response, struct coap_repl
 			msg->send_status_cb(LWM2M_SEND_STATUS_SUCCESS);
 		}
 		return 0;
+	} else if (code == COAP_RESPONSE_CODE_CONTINUE) {
```

Comment:
Shouldn't this already be handled at the CoAP level: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/coap/coap.c#L1744C8-L1744C28. The reply callback should only be called once, for the final block.